packages/client/src/generation/TSClient/Count.ts

```diff
@@ -42,9 +49,21 @@ ${outputType.toTS()}
 export type ${getSelectName(name)} = {
 ${indent(
   type.fields
-    .map(
-      (f) => `${f.name}?: boolean` + (f.outputType.location === 'outputObjectTypes' ? ` | ${getFieldArgName(f)}` : ''),
-    )
+    .map((field) => {
+      const types = ['boolean']
+      if (field.outputType.location === 'outputObjectTypes') {
+        types.push(getFieldArgName(field))
+      }
+
+      // TODO: what should happen if both args and output types are present?
```

Comment:
I don't think that happens now, but it could happen in the future: DMMF definition allows it.